src/modes/draw_line_string.js

```diff
@@ -1,96 +1,101 @@
 const CommonSelectors = require('../lib/common_selectors');
 const LineString = require('../feature_types/line_string');
 const isEventAtCoordinates = require('../lib/is_event_at_coordinates');
-const doubleClickZoom = require('../lib/double_click_zoom');
 const Constants = require('../constants');
 const createVertex = require('../lib/create_vertex');
 
-module.exports = function(ctx) {
-  const line = new LineString(ctx, {
-    type: Constants.geojsonTypes.FEATURE,
-    properties: {},
-    geometry: {
-      type: Constants.geojsonTypes.LINE_STRING,
-      coordinates: []
+import ModeInterface from './mode_interface';
+
+export default class DrawLineStringMode extends ModeInterface {
+  constructor(store, ui) {
+    this.line = new LineString({
+      type: Constants.geojsonTypes.FEATURE,
+      properties: {},
+      geometry: {
+        type: Constants.geojsonTypes.LINE_STRING,
+        coordinates: []
+      }
+    });
+
+    this.currentVertexPosition = 0;
+
+    store.add(this.line);
+    store.clearSelected();
+    this.doubleClickZoom(false);
+    ui.queueMapClasses({ mouse: Constants.cursors.ADD });
+    ui.setActiveButton(Constants.types.LINE);
```

Comment:
So it may be inevitable if we want to allow for custom modes, but this kind of unfortunately balloons the scope of the public API, since now everything in `store` and `ui`, and anything else that gets passed into these methods, is public API, right?